File: Python_Program/pages/home_page.py
from selenium.webdriver.common.by import By
from pages.base_page import BasePage

class HomePage(BasePage):
    CLOSE_LOGIN_POPUP = (By.XPATH, "//button[contains(text(),'✕')]")
    SEARCH_BOX = (By.NAME, "q")

    def close_login_popup(self):
        try:
            self.click(self.CLOSE_LOGIN_POPUP)
        except:
            pass  # Popup not shown

    def search_product(self, product_name):
        self.enter_text(self.SEARCH_BOX, product_name + '\n')

    def verify_title_contains_flipkart(self):
        return "flipkart" in self.get_title().lower()

File: CapstoneProject1/src/main/java/SeleniumTestNG/Pages/HomePage.java
package SeleniumTestNG.Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class HomePage {
	
	private WebDriver driver;

    private By flipkartLogo = By.xpath("//img[@title='Flipkart']");
    private By searchBox = By.name("q");
    private By searchButton = By.xpath("//button[@type='submit']");

    public HomePage(WebDriver driver) {
        this.driver = driver;
    }

    public boolean isFlipkartLogoPresent() {
        try {
            WebElement logo = driver.findElement(flipkartLogo);
            return logo.isDisplayed();
        } catch (Exception e) {
            return false;
        }
    }

    public void enterSearchText(String text) {
        driver.findElement(searchBox).sendKeys(text);
    }

    public void clickSearchButton() {
        driver.findElement(searchButton).click();
    }

    public void searchForItem(String item) {
        enterSearchText(item);
        clickSearchButton();
    }

}
